pilot/pkg/security/authn/v1beta1/policy_applier_test.go

```diff
@@ -691,18 +691,18 @@ func TestJwtFilter(t *testing.T) {
 		},
 	}
 
-	push := model.NewPushContext()
-	push.JwtKeyResolver = model.NewJwksResolver(
-		model.JwtPubKeyEvictionDuration, model.JwtPubKeyRefreshInterval,
-		model.JwtPubKeyRefreshIntervalOnFailure, model.JwtPubKeyRetryInterval)
-	defer push.JwtKeyResolver.Close()
-
-	push.ServiceIndex.HostnameAndNamespace[host.Name("jwt-token-issuer.mesh")] = map[string]*model.Service{}
-	push.ServiceIndex.HostnameAndNamespace[host.Name("jwt-token-issuer.mesh")]["mesh"] = &model.Service{
-		Hostname: "jwt-token-issuer.mesh.svc.cluster.local",
-	}
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
+			push := model.NewPushContext()
```

Comment:
Since we are fetching `pubkey` in the background, the results could be flaky if we create one single object and use it for every test case becuase we don't know when the key will be ready. It's a safe bet to use it inside the for loop and use the `fakePubKey` for every scenario.

Reply:
could you clarify why it is flaky to use one object for every test cases? Isn't that how it is working in real cases? The test should not deviate from the real behavior.

Reply:
all the tests cases separately call the `buildJwksUri()` which calls the `GetOrUpdatePubKey()`. So for the first case it will always return fakejwks, but the other cases with same `jwksuri` and `issuer` will have uncertainty as the method will return fakejwks when cache is empty but if cache got filled in any of the previous cases it will return the real pub key. So we don't exactly know when it will return real pubkey.